src/commands/QueryAdminDetails.tsx

```diff
@@ -0,0 +1,123 @@
+import { Permalinks, UserID, getRequestFn } from "matrix-bot-sdk";
+import { MjolnirContext } from "./CommandHandler";
+import { CommandError, CommandResult } from "./interface-manager/Validation";
+import { defineInterfaceCommand, findTableCommand } from "./interface-manager/InterfaceCommand";
+import { findPresentationType, makePresentationType, parameters, ParsedKeywords, simpleTypeValidator, union } from "./interface-manager/ParameterParsing";
+import "./interface-manager/MatrixPresentations";
+import { JSXFactory } from "./interface-manager/JSXFactory";
+import { defineMatrixInterfaceAdaptor } from "./interface-manager/MatrixInterfaceAdaptor";
+import { renderMatrixAndSend } from "./interface-manager/DeadDocumentMatrix";
+import { DocumentNode } from "./interface-manager/DeadDocument";
+import { ReadItem } from "./interface-manager/CommandReader";
+
+export type MatrixHomeserver = string;
+
+makePresentationType({
+    name: "MatrixHomeserver",
+    // This is a very very crude way to detect a url.
+    validator: simpleTypeValidator("MatrixHomeserver", (readItem: ReadItem) => (readItem instanceof String) && (!readItem.includes('#') || !readItem.includes('!')))
+})
+
+interface SupportJson {
+    contacts?: {
+        matrix_id?: string,
+        email_address?: string;
+        role: "admin" | "security";
+    }[],
+    support_page?: string;
+}
+
+async function queryAdminDetails(
+    this: MjolnirContext,
+    _keywords: ParsedKeywords,
+    entity: UserID | MatrixHomeserver | string
+): Promise<CommandResult<[UserID | MatrixHomeserver | string, SupportJson], CommandError>> {
+    let domain: string;
+    if (entity instanceof UserID) {
+        domain = `https://${entity.domain}`;
+    } else {
+        // Doing some cleanup on the url
+        if (!entity.startsWith("https://") && !entity.startsWith("http://")) {
+            domain = `https://${entity}`;
+        } else {
+            domain = entity;
+        }
+    }
+
+
+    try {
+        const resp: SupportJson = await new Promise((resolve, reject) => {
+            getRequestFn()(`${domain}/.well-known/matrix/support`, (error: any, response: any, resBody: unknown) => {
```

Comment:
![image](https://github.com/the-draupnir-project/Draupnir/assets/50846879/85121d25-a5c0-4ca9-9289-fd6f3c4fc520)

what did i do wrong here? do they just not support the MSC?

Reply:
Ah Hm ok yeah it prints the error but seems like in this case it just printed the whole page oO. Yeah not supported and it went wrong.